packages/blade/src/components/Collapsible/_decisions/decisions.md

```diff
@@ -0,0 +1,186 @@
+# Collapsible
+
+Collapsible is used to allow users to toggle the visibility of hidden content within a container.
+
+This document outlines the API of `Collapsible` component. This shares similarities with [Accordion's API](https://github.com/razorpay/blade/pull/1189).
+
+<div align="center">
+  <img src="./collapsible-thumbnail.png" width="380" alt="" />
+</div>
+
+## Design
+
+- [Figma - Collapsible](https://www.figma.com/file/LSG77hEeVYDk7j7WV7OMJE/Blade-DSL---Components-Guideline?type=design&node-id=79-629874&t=sVxH3DOnx3L3F9rO-0)
+
+## Interactions
+
+Collapsible shares the opening / closing interaction with Accordion:
+
+### Collapsible button
+
+https://github.com/razorpay/blade/assets/6682655/e37daea8-4cbb-42df-b33b-4d32dc0eab65
+
+### Collapsible link
+
+https://github.com/razorpay/blade/assets/6682655/eec42564-4a43-4449-96e1-7c81639d60e7
+
+## API
+
+Annotated component with props ([figma](https://www.figma.com/file/cX625jYMAX1Roq8hLifFrn/Collapsible-breakdown?type=design&node-id=0-1&t=2fpp9pyvcuMxVTjW-0)):
+
+![Collapsible breakdown](./collapsible-breakdown.png)
+
+```jsx
+import { Collapsible, CollapsibleButton, CollapsibleLink, CollapsibleBody } from '@razorpay/blade';
+
+<Collapsible>
+  <CollapsibleButton>Button</CollapsibleButton>
+  <CollapsibleBody>
+    <Box>Slot</Box>
+  </CollapsibleBody>
+</Collapsible>;
+
+<Collapsible>
+  <CollapsibleLink>Button</CollapsibleLink>
+  <CollapsibleBody>
+    <Box>Slot</Box>
+  </CollapsibleBody>
+</Collapsible>;
+```
+
+> **Note**
+>
+> - `Collapsible` will compose (`CollapsibleButton` or `CollapsibleLink`), `CollapsibleBody`. This ensures all interactions, a11y and animations work seamlessly at user end.
+> - `CollapsibleButton` and `CollapsibleLink` will support a subset of relevant props from `Button` and `Link` respectively.
+
+### Collapsible
+
+| Prop              | Type            | Default     | Description                                                        | Required |
+| ----------------- | --------------- | ----------- | ------------------------------------------------------------------ | -------- |
+| children          | `JSX`           | `undefined` | Composes `CollapsibleButton`, `CollapsibleLink`, `CollapsibleBody` | ✅       |
+| direction         | `bottom`, `top` | `bottom`    | Direction in which the content expands                             |          |
+| defaultIsExpanded | `boolean`       | `false`     | Expands the collapsible content by default (uncontrolled)          |          |
+| isExpanded        | `boolean`       | `undefined` | Expands the collapsible content (controlled)                       |          |
+| onExpandChange    | `function`      | `undefined` | Callback for change in collapsible's expanded state                |          |
```

Comment:
should this be `defaultIsOpen`, `isOpen`, `onOpenChange` to [match it with Tooltip](https://github.com/razorpay/blade/pull/1216#discussion_r1220950828)?

Reply:
I feel this is a different kind of interaction and also differs from the behavior. Eg. the content still stays in the DOM (for SEO). Also `expanded` terminology is similar to the `aria-expanded` property which is also for similar disclosures types of interactions.

Reply:
> Eg. the content still stays in the DOM (for SEO)

This is part of internal implementation so I think we can take call irrespective of that. Don't think the consumers would even know about this. We do usually tradeoff "most accurate term" for "consistent term in other components".

> Also expanded terminology is similar to the aria-expanded property which is also for similar disclosures types of interactions.

This is also a good point. "Collapsible is expanded / collapsed" does sound more natural than "Collapsible is open / close" 🤔 

Can take more inputs maybe? what do you think @kamleshchandnani @anuraghazra?

Reply:
Oh just noticed we already discussed it here - https://github.com/razorpay/blade/pull/1218#discussion_r1224073355. Can go with `onExpandChange` then